### DISCUSSION TITLE
Is there any reason why we can't trigger rules both when the transaction is created OR updated?

### DISCUSSION THREAD

**Discussion by tiagodj** [firefly-iii/firefly-iii#7994] [Firefly III]
I have an automated mechanism that creates transactions via API calls.

But sometimes the transactions are not picked up by the rules, and in those cases I manually edit them to match the correct category, budget, etc.

I have rules that sets the correct budget, for example, based on the category. But that rule only is triggered on CREATED transactions.

Which means that when I am editing a transaction that didn't get created in the right category, I also need to manually set the other fields, essentially doing the same work as the rule does.

I was wondering if you could make an option for a rule to trigger when a matching transaction is created and also when it is updated. This would help a lot.

The only current solution for that is to have 2 instances of essentially the same rule, but one for created and one for updated transactions, but this creates a lot of [almost] duplicated rules.

Thanks

**Comment by JC5** [comment]
I know this is an issue but changing it requires some pretty significant database changes and it's not a big enough problem. Sorry about that.

**Reply by tiagodj** [reply]
I understand, thanks!
